pycurl_requests/adapters/pycurl.py

```diff
@@ -83,6 +83,21 @@ def send(self, request, stream=False, timeout=None, verify=True, cert=None, prox
         return pycurl_request.send()
 
 
+class ChunkIterableReader:
+    def __init__(self, iterator: Iterator[bytes]):
+        self._iterator = iterator
+
+    def read(self, ignored) -> bytes:
+        return bytes(next(self._iterator, b""))
+
+    def close(self): # TODO
+        try:
+            self._iterator.close()
+        except AttributeError:
+            pass
```

Comment:
I don't think you should be closing the iterator from within the library. Typically I would expect a user to be using a `with` block or calling `close` manually.